flocker/control/_model.py

```diff
@@ -573,6 +603,17 @@ def __invariant__(self):
                 return (False, '%r is not correct key for %r' % (key, value))
         return (True, "")
 
+    def __new__(cls, **kwargs):
+        # PRecord does some crazy stuff, thus _precord_buckets; see
+        # PRecord.__new__.
+        if "uuid" not in kwargs and "_precord_buckets" not in kwargs:
+            warn("UUID is required, this is for backwards compat with existing"
+                 " tests only. If you see this in production code that's "
+                 "a bug.", DeprecationWarning, stacklevel=2)
+            kwargs["uuid"] = ip_to_uuid(kwargs["hostname"])
+        return PRecord.__new__(cls, **kwargs)
```

Comment:
I'm pretty sure this code is going to be different once the next branch comes around (going to remove hostname attribute from `Node`). So if it's still duplicative I'll fix it there.